readthedocs/projects/forms.py

```diff
@@ -357,6 +358,26 @@ def clean_conf_py_file(self):
             )  # yapf: disable
         return filename
 
+    def clean_build_config_file(self):
+        """
+        Validate user input to help user.
+
+        We also validate this path before using it to read a file, so this is
+        only considered helpful to a user.
+        """
+        filename = self.cleaned_data.get("build_config_file")
+        filename = (filename or "").strip()
+        if filename.startswith("/"):
+            raise forms.ValidationError(
+                _(
+                    "Use a relative path. It should not begin with '/'. "
+                    "The path is relative to the root of your repository."
+                )
+            )
+        if ".." in filename:
+            raise forms.ValidationError(_("Found invalid sequence in path: '..'"))
```

Comment:
This seems like a small set of the actual checks if the goal is security? @stsewd might have some thoughts on path security stuff. Should we have a standard way to check this?

Reply:
This is NOT security, that was also what the docstring was there to clarify. I'd never consider this security, just ensuring that this string pattern doesn't occur in the input. We have actual security to sandbox builds :)

Reply:
I really do like the idea of standard "path" validation and [asked for it earlier](https://github.com/readthedocs/readthedocs.org/pull/10001#discussion_r1132650217). It could make sense that we write _our own_ simply because of being able to raise `ValidationError` with helpful messages + because we might have tighter rules than a normal unix path.

I'll do that for this one, I can see there is an objection to changing the validation of the Sphinx conf.py path :)

Reply:
Added a reusable validation pattern and validation of the validation.

Would like to note that I DID look online if there was something that looked quickly reusable and there wasn't.